### PR TITLE
introduce `helm init --automount-service-account-token`

### LINKED ISSUE #2464: Tiller pods can't connect to k8s apiserver
logs:
```
(kube-master) tiller # kcm logs tiller-deploy-3354596499-4q16f
Cannot initialize Kubernetes connection: Get http://localhost:8080/api: dial tcp [::1]:8080: getsockopt: connection refused
```
This issue has been raised in #1591 #1791.

In #1791, there gets a resolution(thank @iamzhout) of adding **KUBERNETES_MASTER** in pods env  but nobody explains the root cause of the issue.

I installed the k8s cluster manually and it seems that  tiller pod can't get **KUBERNETES_MASTER** from env as this item doesn't exist in the **env** at all. 

No docs are found about the relation between k8s cluster and the configuration.  I don't  think it is a good idea to add the env item manually(and how to `helm init` with the configuration???)

Is there anybody who can offer the help? 

Thanks anyway (^.^).

### PR DESCRIPTION
**Pull request by bacongobbler**
supercedes #4229 by making the flag configurable, but enabled by default.

closes #2464.

### CODE REVIEW

**Comment by jascott1** [comment]
@bacongobbler Looking good. Does this need more documentation or do you think we are good?

**Comment by bacongobbler** [comment]
ok, I added a quick one-liner for this feature flag. Honestly there's not much reason why you'd want to disable this flag so there's not much to write home about this flag, but there it is. ;)

File: cmd/helm/installer/install_test.go [hunk 1]
```diff
@@ -80,14 +80,27 @@ func TestDeploymentForServiceAccount(t *testing.T) {
...
+		d, err := Deployment(opts)
 		if err != nil {
 			t.Fatalf("%s: error %q", tt.name, err)
 		}
 
 		if got := d.Spec.Template.Spec.ServiceAccountName; got != tt.serviceAccount {
 			t.Errorf("%s: expected service account value %q, got %q", tt.name, tt.serviceAccount, got)
 		}
+		if got := *d.Spec.Template.Spec.AutomountServiceAccountToken; got != false {
+			t.Errorf("%s: unexpected automountServiceAccountToken = %t", tt.name, got)
```
Comment (jascott1) [inline cmd/helm/installer/install_test.go]: You might consider failing with the "expected value X got value Y format" to make the expected result clear.

**Review by jascott1 — COMMENTED** [review]
Optional nits on the test verbiage.

**Comment by bacongobbler** [comment]
done

**Review by jascott1 — APPROVED** [review]

**Pull request merged**